website/docs/r/kubernetes_cluster_node_pool_snapshot.html.markdown

```diff
@@ -0,0 +1,96 @@
+---
+subcategory: "Container"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_kubernetes_cluster_node_pool_snapshot"
+description: |-
+  Manages a Kubernetes Cluster Node Pool Snapshot.
+
+---
+
+# azurerm_kubernetes_cluster_node_pool_snapshot
+
+Manages a Kubernetes Cluster Node Pool Snapshot
```

Comment:
@heiazuo what does this resource _actually_ do?

Reply:
> Node pool snapshots allow you to take a configuration snapshot of your node pool and then create new node pools or new clusters based of that snapshot for as long as that configuration and kubernetes version is supported.

refs: https://docs.microsoft.com/en-us/azure/aks/node-pool-snapshot

Reply:
So from a usage perspective, users would capture an image of the node pool, take a snapshot, delete the original node pool, then provision another? From a Terraform perspective, since the `node_pool_id` field is ForceNew, removing the Node Pool (or this field from the Resource) would destroy/recreate this - so as it stands this resource isn't that useful / will need rethinking to account for that - can you take another look into how this would work end-to-end?

Thanks

Reply:
Maybe user can construct the node pool id instead of referring a node pool resource? Then user can copy the existing node pool's configuration and create a new one.